examples/3-adapter_advanced.ipynb

```diff
@@ -0,0 +1,740 @@
+   "metadata": {},
+   "source": [
+    "We define `RandomForestTransformer` in similar manner as before. With one difference, though. `Transform` will use RandomForest's `predict_proba` instead of `predict` which will be useful in the latter part of this notebook."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class RandomForestTransformer(BaseTransformer):\n",
+    "    def __init__(self):\n",
+    "        self.estimator = RandomForestClassifier(n_estimators=10, max_depth=10, random_state=12345)\n",
+    "        \n",
+    "    def fit(self, X, y):\n",
+    "        self.estimator.fit(X, y)\n",
+    "        return self\n",
+    "\n",
+    "    def transform(self, X, **kwargs):\n",
+    "        y_proba  = self.estimator.predict_proba(X)\n",
+    "        return {'y_proba': y_proba}\n",
+    "    \n",
+    "    def save(self, filepath):\n",
+    "        joblib.dump(self.estimator, filepath)\n",
+    "        \n",
+    "    def load(self, filepath):\n",
+    "        self.estimator = joblib.load(filepath)\n",
+    "        return self"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "rf_step = Step(name='random_forest',\n",
+    "               transformer=RandomForestTransformer(),\n",
+    "               input_data=['input'],        \n",
+    "               cache_dirpath=CACHE_DIR,\n",
+    "               force_fitting=True)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "rf_step"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The graph looks just like in notebook #1. Let's try to execute it!"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "try:\n",
+    "    preds_train_rf = rf_step.fit_transform(data_train)\n",
+    "except:\n",
+    "    traceback.print_exc()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As we can see, something went wrong. The problem is that `input` dictionary in `data_train` contains fields `images` and `labels`, whereas `RandomForestTransformer` expects arguments `X` and `y`."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## The solution: adapter"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To handle such issues, `Step`'s initializer has `adapter` argument. `Adapter` describes how to reshape the data from the input nodes into the form expected by the transformer or further steps. \n",
+    "\n",
+    "The basic usage is as follows:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "rename_step = Step(name='rename',\n",
```

Comment:
Maybe `renamed_step`?

Reply:
(instead of `rename_step`)

Reply:
Never mind, I didn't notice that this step is a dummy renaming step. Please ignore.